Input with label

HagerRagaee opened the pull request.

HagerRagaee committed:
InputWithlabel implementation

HagerRagaee committed:
inputWithLabel

Ofahmy143 commented:
Branch name should be feature/{feature-name}

Ahmed-Khaled24 commented on src/components/InputWithLabel.jsx:
```diff
@@ -0,0 +1,30 @@
+            <div className="input">
+                <label className="label" >{props.Labelvalue}</label>
+                <form >
+                    <input type="text" placeholder={"Enter " + props.Labelvalue}/>
```
It would be better if we pass a custom placeholder prop instead of use the label as the placeholder

Ahmed-Khaled24 commented on src/components/InputWithLabel.jsx:
```diff
@@ -0,0 +1,30 @@
+import "./inputWithLabel.css"
+function InputWithLabel(props) {
+    const type = props.Type;
+    if (type == "input") {
```
using the string `input` as a test variable is more error-prone, so we can use a boolean variable for example `multiline` if it is true we return the textarea and else we return input.

Ahmed-Khaled24 commented on src/components/InputWithLabel.jsx:
```diff
@@ -0,0 +1,30 @@
+        return (
+            <div className="input">
+                <label className="label" >{props.Labelvalue}</label>
+                <form >
+                    <input type="text" placeholder={"Enter " + props.Labelvalue}/>
+                </form>
```
Do we have to use this form element?

Ahmed-Khaled24 commented on src/components/InputWithLabel.jsx:
```diff
@@ -0,0 +1,30 @@
+import "./inputWithLabel.css"
+function InputWithLabel(props) {
+    const type = props.Type;
```
we may use object destructureing here for cleaner code and it would help if we pass other props.

HagerRagaee committed:
input field with changes

HagerRagaee commented:
all changes done.
remove unnecessary form, add custom placeholder prop instead of use the label as the placeholder, use bool var instead of string Type

according to name of branch i can change name to input field??

Ahmed-Khaled24 commented on src/components/inputWithLabel.css:
## Styling doesn't match the design 

### This is yours 
![image](https://github.com/ASU-Semicolon/The-New-Dashboard/assets/80486237/ad07912a-feae-47a1-95b0-af7dac129021)

### And this is the design
![image](https://github.com/ASU-Semicolon/The-New-Dashboard/assets/80486237/a43ff33e-879f-4d2b-a117-3b96fb6762ca)

### Other comments
- It is better to use `rem` over `px` 
- Always name classes with meaningful names to make it easier to understand

Ahmed-Khaled24 commented on src/components/InputWithLabel.jsx:
```diff
@@ -0,0 +1,25 @@
+import "./inputWithLabel.css"
+function InputWithLabel({ Labelvalue, placeholder, multiline }) {
```
please use the same casing to name all variables `Labelvalue` needs to be either `labelValue` or just `label` as it is self-explanatory

Ahmed-Khaled24 commented on src/components/InputWithLabel.jsx:
```diff
@@ -0,0 +1,25 @@
+import "./inputWithLabel.css"
+function InputWithLabel({ Labelvalue, placeholder, multiline }) {
+    if (!multiline) {
+        return (
+            <div className="input">
+                <label className="label">{Labelvalue}</label>
+                <input type="text" placeholder={placeholder}/>
+            </div>
+        );
+    } else {
+        return (
+            <div className="input">
+                <label className="label">{Labelvalue}</label>
+                <textarea placeholder={placeholder} />
+            </div>
+        );
+    }
+}
```
There is code duplication here always stick to this principle `Don't repeat yourself`
Here is a better way to write this 
```js
	return (
		<div className="input">
			<label className="label">{Labelvalue}</label>
			{multiline ? (
				<textarea placeholder={placeholder} />
			) : (
				<input type="text" placeholder={placeholder} />
			)}
		</div>
	);
```

Ahmed-Khaled24 reviewed (changes requested):
As we discussed in the last meeting 
component file has extension .component.jsx
styles file has extension .style.css 
and both added inside a folder with `kebab-case` name of the component 

Good job and waiting for your new changes.

HagerRagaee committed:
feat: Improve styling and readability of InputWithLabel component

- Changed pixel values to rem units for better responsiveness.
- Styled label text color to white for better contrast.
- Adjusted placeholder font size and family.
- Ensured consistent border styling for input fields.
